crates/rpc/rpc/src/eth/error.rs

```diff
@@ -24,13 +24,48 @@ pub(crate) enum EthApiError {
 impl_to_rpc_result!(EthApiError);
 
 /// A helper error type that ensures error messages are compatible with `geth`
-// TODO: replace thiserror with custom convert
 #[derive(Debug, thiserror::Error)]
 #[error(transparent)]
 pub(crate) struct GethCompatPoolError(PoolError);
 
+impl GethCompatPoolError {
+
+    fn from(err: PoolError) -> GethTxPoolError {
+        match err {
+            PoolError::ReplacementUnderpriced(_) =>GethTxPoolError::ReplaceUnderpriced,
+            PoolError::ProtocolFeeCapTooLow(_, _) => GethTxPoolError::Underpriced,
+            PoolError::SpammerExceededCapacity(_, _) => GethTxPoolError::TxPoolOverflow,
+            PoolError::DiscardedOnInsert(_) => GethTxPoolError::TxPoolOverflow,
+            PoolError::TxExceedsGasLimit(_, _, _) => GethTxPoolError::GasLimit,
+            PoolError::TxExceedsMaxInitCodeSize(_, _, _) => GethTxPoolError::OversizedData,
+        }
+    }
+
+}
+
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum GethTxPoolError {
```

Comment:
this is great!

now that we have this we could get rid of `GethCompatPoolError` entirely and impl

`From<PoolError> for EthApiError`
`From<PoolError> for EthApiError`

so we can use the `?` directly.